src/main/java/net/snowflake/ingest/streaming/internal/StreamingIngestUtils.java

```diff
@@ -77,7 +80,7 @@ public static void sleepForRetry(int executionCount) {
   static <T extends StreamingIngestResponse> T executeWithRetries(
       Class<T> targetClass,
       String endpoint,
-      Map<Object, Object> payload,
+      StreamingIngestRequest payload,
```

Comment:
since its an empty interface, unless you plan to add something to that common interface you can also do the following:

```
 static <TRequest, TResponse extends StreamingIngestResponse> TResponse executeWithRetries(
      Class<TResponse> targetClass,
      String endpoint,
      TRequest payload,
      String message,
      ServiceResponseHandler.ApiName apiName,
      CloseableHttpClient httpClient,
      RequestBuilder requestBuilder)
      throws IOException, IngestResponseException {
```

Reply:
closing this thread out, just added a comment where i said we should add a common toStringForLogging() method to all request classes (and add this to the interface too)